index.js

```diff
@@ -128,40 +148,36 @@ exports.empty_return_path = function (next, connection) {
   // validate that the Return-Path header is empty, RFC 3834
 
   const rp = transaction.header.get('Return-Path')
-  if (!rp) {
+  if (!rp || rp === '<>') {
     transaction.results.add(this, { pass: 'empty_return_path' })
     return next()
   }
 
-  if (rp === '<>') {
-    transaction.results.add(this, { pass: 'empty_return_path' })
-    return next()
+  transaction.results.add(this, { fail: 'empty_return_path', emit: true })
+
+  if (this.cfg.reject.empty_return_path) {
+    return next(DENY, 'bounce with non-empty Return-Path (RFC 3834)')
   }
 
-  transaction.results.add(this, { fail: 'empty_return_path', emit: true })
-  return next(DENY, 'bounce with non-empty Return-Path (RFC 3834)')
+  next()
 }
 
-exports.bad_rcpt = function (next, connection) {
-  if (!this.cfg.check.bad_rcpt) return next()
+exports.bad_rcpt = function (next, connection, rcpt) {
+  if (!this.cfg.reject.bad_rcpt) return next()
   if (!this.has_null_sender(connection)) return next()
-  if (!this.cfg.invalid_addrs) return next()
 
-  const { transaction } = connection
-  for (const element of transaction.rcpt_to) {
-    const rcpt = element.address()
-    if (!this.cfg.invalid_addrs[rcpt]) continue
-    transaction.results.add(this, { fail: 'bad_rcpt', emit: true })
+  if (this.cfg.invalid_addrs.includes(rcpt.address())) {
+    connection.transaction.results.add(this, { fail: 'bad_rcpt', emit: true })
     return next(DENY, 'That recipient does not accept bounces')
   }
-  transaction.results.add(this, { pass: 'bad_rcpt' })
+  connection.transaction.results.add(this, { pass: 'bad_rcpt' })
 
   next()
 }
 
 exports.has_null_sender = function (connection, mail_from) {
-  const transaction = connection?.transaction
-  if (!transaction) return false
+  const { transaction } = connection
+  if (!transaction) return next()
```

Comment:
```suggestion
  if (!transaction) return false
```

There's no `next()` in context here.